lib/Basics/debugging.cpp

```diff
@@ -121,6 +126,7 @@ void TRI_AddFailurePointDebugging(std::string_view value) {
   }
 
   if (added) {
+    ::hasFailurePoints.store(true, std::memory_order_relaxed);
```

Comment:
Updating the list of failure points and the new atomic variable does not happen atomically.
So this will introduce a potential race in a multi-threaded setup, because we are updating the atomic variable while not holding the lock.
This is likely not a problem for our current tests which set a failure point via this API and wait until the API call returns. But for other multi-threaded tests it may cause issues (in the future).

Reply:
Fixed!